charms/kserve-controller/src/charm.py

```diff
@@ -36,28 +41,36 @@
     "src/templates/auth_manifests.yaml.j2",
     "src/templates/serving_runtimes_manifests.yaml.j2",
     "src/templates/webhook_manifests.yaml.j2",
-    "src/templates/configmap_manifests.yaml.j2",
 ]
-CONFIG_FILE = "src/config/inference-config.j2"
+CONFIG_FILES = ["src/templates/configmap_manifests.yaml.j2"]
 
 
 class KServeControllerCharm(CharmBase):
     """Charm the service."""
 
     def __init__(self, *args):
         super().__init__(*args)
+        self._ingress_gateway_requirer = GatewayRequirer(self, relation_name="ingress-gateway")
+        self._local_gateway_requirer = GatewayRequirer(self, relation_name="local-gateway")
 
         self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.remove, self._on_remove)
         self.framework.observe(
             self.on.kserve_controller_pebble_ready, self._on_kserve_controller_ready
         )
         self.framework.observe(
             self.on.kube_rbac_proxy_pebble_ready, self._on_kube_rbac_proxy_ready
         )
-
+        self.framework.observe(
+            self.on["ingress-gateway"].relation_changed, self._on_ingress_gateway_relation_changed
+        )
+        self.framework.observe(
+            self.on["local-gateway"].relation_changed, self._on_local_gateway_relation_changed
+        )
         self._k8s_resource_handler = None
```

Comment:
When relation is removed (broken or departed) should we have a special handling that puts charm back into Blocked state?

Reply:
Yeah, good point.

Reply:
Updated it, please let me know what you think.

Reply:
Perfect!